python/python_fastapi_server/routers/edr.py

```diff
@@ -0,0 +1,876 @@
+        interval = [[
+            datetime.strptime(time_dim["values"][0], "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc),
+            datetime.strptime(time_dim["values"][-1],
+                                "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc),
+        ]]
+        return interval, time_dim["values"]
+    return (None, None)
+
+
+def get_custom_dims_for_collection(edr_collectioninfo: dict, parameter: str = None):
+    wms = get_capabilities(edr_collectioninfo['name'])
+    custom = []
+    if parameter and parameter in list(wms):
+        layer = wms[parameter]
+    else:
+        # default to first layer
+        layer = wms[list(wms)[0]]
+    for dim_name in layer.dimensions:
+        if dim_name not in ["reference_time", "time", "elevation", edr_collectioninfo.get("vertical_name")]:
+            custom_dim = {
+                "id": dim_name,
+                "interval": [],
+                "values": layer.dimensions[dim_name]["values"],
+                "reference": f"custom_{dim_name}",
+            }
+            custom.append(custom_dim)
+    return custom if len(custom) > 0 else None
+
+
+def get_vertical_dim_for_collection(edr_collectioninfo: dict, parameter: str = None):
+    wms = get_capabilities(edr_collectioninfo['name'])
+    if parameter and parameter in list(wms):
+        layer = wms[parameter]
+    else:
+        layer = wms[list(wms)[0]]
+
+    for dim_name in layer.dimensions:
+        if dim_name in ["elevation"] or ( "vertical_name" in edr_collectioninfo and dim_name==edr_collectioninfo["vertical_name"]):
+            vertical_dim = {
+                "interval": [],
+                "values": layer.dimensions[dim_name]["values"],
+                "vrs": "customvrs",
+            }
+            return vertical_dim
+    return None
+
+
+@edrApiApp.get("/collections",
+               response_model=Collections,
+               response_model_exclude_none=True)
+async def edr_get_collections(request: Request):
+    links: list[Link] = []
+    base_url = str(request.url_for("edr_get_collections"))
+    self_link = Link(href=base_url, rel="self", type="application/json")
+
+    links.append(self_link)
+    collections: list[Collection] = []
+    edr_collections = get_edr_collections()
+    for edr_coll in edr_collections:
+        coll = get_collectioninfo_for_id(edr_coll)
+        collections.append(coll)
+    collections_data = Collections(links=links, collections=collections)
+    return collections_data
+
+
+@edrApiApp.get("/collections/{collection_name}",
+               response_model=Collection,
+               response_model_exclude_none=True)
+async def edr_get_collection_by_id(collection_name: str, request: Request):
+    base_url = str(request.url_for("edr_get_collection_by_id",
+                               collection_name=collection_name))
+
+    collection = get_collectioninfo_for_id(collection_name)
+
+    return collection
+
+
+@cached(cache=TTLCache(maxsize=1024, ttl=60))
```

Comment:
So after one minute the cache gets refreshed?  Something to mention in the docs, as other changes to dataset configurations are instantaneous. Changes to the EDR section are then cached for 1 minute.